scripts/Inelastic/CrystalField/CrystalFieldMultiSite.py

```diff
@@ -0,0 +1,533 @@
+        """Form a definition string for the CrystalFieldSpectrum function
+        @param i: Index of a spectrum.
+        """
+        if self.NumberOfSpectra == 1:
+            return str(self.function)
+        else:
+            funs = self.function.createEquivalentFunctions()
+            return str(funs[i])
+
+    def _makeAbundances(self, abundances):
+        """Create dict for ion intensity scalings"""
+        if abundances is not None:
+            for ion_index in range(len(self.Ions)):
+                self._abundances['ion%s' % ion_index]  = abundances[ion_index]
+            max_ion = max(self._abundances, key=lambda key: self._abundances[key])
+            ties = {}
+            for ion in self._abundances.keys():
+                if ion is not max_ion:
+                    factor = self._abundances[ion] / self._abundances[max_ion]
+                    tie_key = ion + '.IntensityScaling'
+                    tie_value = str(factor) + '*' + max_ion + ".IntensityScaling"
+                    ties[tie_key] = tie_value
+                    self.ties(ties)
+        else:
+            for ion_index in range(len(self.Ions)):
+                self._abundances['ion%s' % ion_index]  = 1.0
+
+    def update(self, func):
+        """
+        Update values of the fitting parameters.
+        @param func: A IFunction object containing new parameter values.
+        """
+        self.function = func
+
+    def fix(self, *args):
+        for a in args:
+            self.function.fixParameter(a)
+
+    def ties(self, *args, **kwargs):
+        """
+        Set ties on the field parameters.
+
+        @param kwargs: Ties as name=value pairs: name is a parameter name,
+            the value is a tie string or a number. For example:
+                tie(B20 = 0.1, IB23 = '2*B23')
+        """
+        for arg in args:
+            if isinstance(arg, dict):
+                kwargs.update(arg)
+            else:
+                raise TypeError("")
+        for tie in kwargs:
+            self.function.tie(tie, str(kwargs[tie]))
+
+    def constraints(self, *args):
+        """
+        Set constraints for the field parameters.
+
+        @param args: A list of constraints. For example:
+                constraints('B00 > 0', '0.1 < B43 < 0.9')
+        """
+        self.function.addConstraints(','.join(args))
+
+    def plot(self, i=0, workspace=None, ws_index=0, name=None):
+        """Plot a spectrum. Parameters are the same as in getSpectrum(...)"""
+        from mantidplot import plotSpectrum
+        from mantid.api import AlgorithmManager
+        createWS = AlgorithmManager.createUnmanaged('CreateWorkspace')
+        createWS.initialize()
+
+        xArray, yArray = self.getSpectrum(i, workspace, ws_index)
+        ws_name = name if name is not None else 'CrystalFieldMultiSite_%s' % self.Ions
+
+        if isinstance(i, int):
+            if workspace is None:
+                if i > 0:
+                    ws_name += '_%s' % i
```

Comment:
I don't understand, why is ^ better than 
```
if i > 0:
    ws_name += '_{}'.format(i)
```

Reply:
tbh there's not much in it. Writing it as a one liner saves an extra line & an indent. The above is a little more readable. I noticed you'd already used ternary operators in this function so thought it might be more consistent. Up to you.